### ISSUE TITLE
FP S1172: private tuple parameter usage not regonized

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#3838]
### Description
I saw your pull request #3255 , but I'm not sure if it is the same issue. (You might want to add these tests anyway. So this Tuple is not recognized when part of a private function, contrary to the public one. (In that pull request is seems to be an issue with public methods to, so that why I guess it might be a different issue).

``` C#
using System.Collections.Generic;
using System.Linq;

namespace SonarAnalyzer.Experiments.CSharp
{
    public static class S1172
    {
        public static IEnumerable<(int, int)> Pairs()
        {
            return Enumerable.Range(0, 1).Select(i => PrivatePair(i, i * i));
        }

        private static (int, int) PrivatePair(int l, int r) => (l, r); // compliant FP
        public static (int, int) PublicPair(int l, int r) => (l, r); // compliant - OK already.
    }
}
```

### Related information
* 8.15.0.24505

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Corniel ,

Thank you for reporting this case. It was indeed fixed by the PR #3255 and you should get the fix with the next release!

The public/private case was just my confusion about the rule intention. It only raises on methods that [can be safely changed](https://github.com/SonarSource/sonar-dotnet/blob/ad563ddbbea856ec049b94c189800c4693ad1033/analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnused.cs#L193), therefore private ones.
